client/my-sites/checklist/wpcom-checklist/component.jsx

```diff
@@ -478,6 +479,47 @@ class WpcomChecklistComponent extends PureComponent {
 			/>
 		);
 	};
+
+	renderEmailSetupTask = ( TaskComponent, baseProps, task ) => {
+		const { translate, siteSlug } = this.props;
+
+		const emailSetupProps = {
+			title: translate( 'Get email for your site' ),
+			bannerImageSrc: '/calypso/images/stats/tasks/email.svg', // TODO: get an actual svg for this
+			description: translate(
+				'Subscribe to G Suite to get a decidated inbox, docs, and cloud storage.'
+			),
+			duration: translate( '%d minute', '%d minutes', { count: 5, args: [ 5 ] } ),
+			onClick: () => {
+				page( `/domains/manage/email/${ siteSlug }` );
+			},
+			onDismiss: () => this.handleTaskDismiss( task.id ),
+		};
+
+		const emailForwardingUsed = {
+			completedTitle: translate( 'You set up email forwarding for your site' ),
+			completedDescription: translate(
+				'Want a dedicated inbox, docs, and cloud storage? {{link}}Upgrade to G Suite!{{/link}}',
+				{
+					components: {
+						link: <a href={ `/domains/manage/email/${ siteSlug }` } />,
+					},
+				}
+			),
+			completedButtonText: translate( 'Upgrade' ),
+		};
+		const emailForwardingNotUsed = {
+			completedTitle: translate( 'You set up email for your site' ),
+		};
+
+		const emailForwardingProps = task.email_forwarding
+			? emailForwardingUsed
+			: emailForwardingNotUsed;
+
+		const props = Object.assign( {}, baseProps, emailSetupProps, emailForwardingProps );
+
+		return <TaskComponent { ...props } />;
```

Comment:
Since that part is JSX instead of ES6, it had to be like this but it is definitely cleaner than before.

``` javascript
<TaskComponent { ...baseProps } { ...emailSetupProps } { ...emailForwardingProps } />
```